torch/fx/experimental/symbolic_shapes.py

```diff
@@ -265,6 +266,29 @@ def check_consistent(new, old) -> None:
         assert isinstance(old, scalar_types) and not isinstance(old, bool), f"{old} != {new}"
         torch._check(old == new, lambda: f"{old} != {new} (old != new)")
 
+def rebind_unbacked(shape_env, n: torch.fx.Node, result):
+    """
+    Suppose we are retracing a pre-existing FX graph that previously had
+    fake tensor propagation (and therefore unbacked SymInts).  When we retrace,
+    we re-propagate fake tensors, which results in new unbacked SymInts.
+    When this happens, we need to tell the shape environment about the equivalence
+    of the old and new unbacked SymInts.  Pass us the old torch.fx.Node (which
+    has the old binding information) and the new result (which we can extract the
+    new unbacked SymInts out from).
+    """
+    if bindings := n.meta.get("unbacked_bindings"):
+        for raw_u0, path in bindings.items():
+            u1 = pytree.key_get(result, path)
+            # We should never have bindings for raw bools; instead they should
+            # have been converted to ints via ConvertIntKey
+            assert type(u1) is not bool
```

Comment:
maybe I'm holding it wrong - but shouldn't this assert be against `SymBool` (the symbolic expression stored on the FX node), not an actual bool/int/float?

Reply:
You are indeed right, and I end up having to fix this code better at https://github.com/pytorch/pytorch/pull/124394/files#diff-03557db7303b8540f095b4f0d9cd2280e1f42f534f67d8695f756ec6c02d3ec7L282-R311 later in the stack (the problem doesn't manifest until the inductor tests are online though)